### PR TITLE
get report rest api

### PR DESCRIPTION
**Pull request by haonliu**
#### Description of changes

(Give an overview. Add a technical description describing your changes.)

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Added relevant unit test for your changes. (`yarn test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [ ] Ran precheckin (`yarn precheckin`)

### CODE REVIEW

File: packages/web-api/src/controllers/scan-report-controller.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    ) {
+        super();
+    }
+
+    public async handleRequest(): Promise<void> {
+        const reportId = <string>this.context.bindingData.reportId;
+        if (!this.guidGenerator.isValidV6Guid(reportId)) {
+            this.context.res = {
+                status: 422,
+                body: `Unprocessable Entity: ${reportId}.`,
```
Comment (manishsat) [inline packages/web-api/src/controllers/scan-report-controller.ts]: Change to "invalid report id" or something.

Comment (haonliu) [reply packages/web-api/src/controllers/scan-report-controller.ts]: done.

File: packages/web-api/src/controllers/scan-report-controller.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        if (blobContentDownloadResponse.notFound === true) {
+            this.context.res = {
+                status: 404,
+            };
+
+            return;
+        }
+
+        const stream = new Stream.PassThrough();
+        blobContentDownloadResponse.content.pipe(stream);
```
Comment (devaradhanm) [inline packages/web-api/src/controllers/scan-report-controller.ts]: can't we directly pass the stream returned by blobContentDownloadResponse.content?

Comment (devaradhanm) [reply packages/web-api/src/controllers/scan-report-controller.ts]: Or is this the pattern when dealing with rest api response?

Comment (flyingUnderTheRadar) [reply packages/web-api/src/controllers/scan-report-controller.ts]: With [other frameworks](https://dev.azure.com/mseng/_git/AzureDevOps?path=%2FKeros%2FService%2FServer%2FUtility%2FHttpResponseMessageCreator.cs&version=GBmaster) it is possible to send a stream. Not sure whether we can do it with typescript. We can also punt this to a bug.

Comment (haonliu) [reply packages/web-api/src/controllers/scan-report-controller.ts]: I passed content as body directly initially but the client never get response.

File: packages/web-api/src/controllers/scan-report-controller.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+        if (blobContentDownloadResponse.notFound === true) {
+            this.context.res = {
+                status: 404,
+            };
+
+            return;
+        }
+
+        const content = await getRawBody(blobContentDownloadResponse.content as Readable);
```
Comment (devaradhanm) [inline packages/web-api/src/controllers/scan-report-controller.ts]: have you validated if this works & u are able to read the content?

**Pull request merged**